tests/integration/test_charm.py

```diff
@@ -16,35 +14,66 @@
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 APP_NAME = METADATA["name"]
 
+INI_PATH = "/etc/pgbouncer/pgbouncer.ini"
+USERLIST_PATH = "/etc/pgbouncer/userlist.txt"
+
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
+    """Build and deploy pgbouncer charm."""
+    charm = await ops_test.build_charm(".")
+    resources = {
+        "pgbouncer-image": METADATA["resources"]["pgbouncer-image"]["upstream-source"],
+    }
+    await ops_test.model.deploy(
+        charm,
+        resources=resources,
+        application_name=APP_NAME,
+        config={"pgb_admin_users": "juju-admin"},
+    )
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
```

Comment:
might want a test for reload_pgbouncer_action .

Reply:
actually nvm I see this feature isn't implemented and you have `pass` for it in your unit tests. See my comment below :)